Source/driver/sdc_util.F90

```diff
@@ -103,248 +222,144 @@ subroutine sdc_solve(dt_m, U_old, U_new, C, sdc_iteration)
+    ! temperature will be used as an initial guess in the EOS
+    rpar(irp_temp) = U_old(UTEMP)
 
-             atol_spec(:) = merge(abs(dU_react(1:nspec_evolve)), ZERO, &
-                                  abs(dU_react(1:nspec_evolve)) > sdc_solver_atol * U_react(0))
+    ! we should be able to do an update for this somehow?
+    if (sdc_solve_for_rhoe == 1) then
+       rpar(irp_evar) = U_new(UEDEN)
+    else
+       rpar(irp_evar) = U_new(UEINT)
+    endif
 
-             ! construct the norm of the correction
-             w(0) = abs(dU_react(0)/(U_react(0) + SMALL_X_SAFE))
-             w(1:nspec_evolve) = abs(atol_spec(1:nspec_evolve)/(U_react(1:nspec_evolve) + SMALL_X_SAFE))
-             w(nspec_evolve+1) = abs(dU_react(nspec_evolve+1)/(U_react(nspec_evolve+1) + SMALL_X_SAFE))
+    rpar(irp_spec:irp_spec-1+(nspec-nspec_evolve)) = &
+         U_new(UFS+nspec_evolve:UFS-1+nspec)
 
-             err_dens = abs(w(0))
-             err_spec = maxval(w(1:nspec_evolve))
-             err_ener = abs(w(nspec_evolve+1))
+    ! store the subset for the nonlinear solve
+    ! We use an initial guess if possible
+    U_react(0) = U_new(URHO)
+    U_react(1:nspec_evolve) = U_new(UFS:UFS-1+nspec_evolve)
+    if (sdc_solve_for_rhoe == 1) then
+       U_react(nspec_evolve+1) = U_new(UEINT)
+    else
+       U_react(nspec_evolve+1) = U_new(UEDEN)
+    endif
 
-             if (err_dens < tol_dens .and. err_spec < tol_spec .and. err_ener < tol_ener) then
-                converged = .true.
-             endif
+#if (INTEGRATOR == 0)
+    ! do a simple Newton solve
 
-             iter = iter + 1
-          enddo
+    ! iterative loop
+    iter = 0
+    max_newton_iter = MAX_ITER
 
-          if (converged) then
-             exit
-          endif
+    err = 1.e30_rt
+    converged = .false.
+    do while (.not. converged .and. iter < max_newton_iter)
 
-          if (.not. converged) then
-             if (sdc_solver == 4) then
-                ! we didn't converge, so let's try VODE next
-                solver = VODE_SOLVE
-                cycle
-             endif
+       call f_sdc_jac(nspec_evolve+2, U_react, f, Jac, nspec_evolve+2, info, rpar)
 
-             print *, "errors: ", err_dens, err_spec, err_ener
-             call amrex_error("did not converge in SDC")
-          endif
+       ! solve the linear system: Jac dU_react = -f
+       call dgefa(Jac, nspec_evolve+2, nspec_evolve+2, ipvt, info)
+       if (info /= 0) then
+          ierr = SINGULAR_MATRIX
+          return
+       endif
 
-       else if (solver == VODE_SOLVE) then
+       f_rhs(:) = -f(:)
 
-          ! use VODE to do the solve
+       call dgesl(Jac, nspec_evolve+2, nspec_evolve+2, ipvt, f_rhs, 0)
 
-          istate = 1
-          iopt = 1
+       dU_react(:) = f_rhs(:)
 
-          iwork(:) = 0
+       ! how much of dU_react should we apply?  We don't want species
+       ! going negative.  Here, the 1/2 is a safety factor
+       eta = ONE
+       ! do n = 1, nspec_evolve
```

Comment:
Is this part of the code still a work in progress for future PRs, or do you think we're doing the right thing here now?

Reply:
I think that was an experiment that I no longer need